### PR TITLE
feat: handle buffered events

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

CoreCrypo 1.0.0 introduced a new feature where it will buffer message received after joining a conversation via external commit but before `mergePendingGroupFromExternalCommit` has been called. This scenario will happen if an MLS group has any pending external remove proposals when joining via external commit. The reason is that the backend will immediately re-create the remove proposals because they get invalidated by the external commit.

This PR updates `JoinExistingMLSConversationUseCase` and `JoinSubconversationUseCase` to process any buffered proposals.

#### Test Coverage

- [x] I have added automated test to this contribution

#### How to Test

1. Create MLS group with user A and B
2. Make user A's client is offline so it can process proposals 
3. Leave the group with user B's client (the group will now have pending remove proposal)
4. Login with a second client for user A (after login a client will join all known MLS groups via external commit)

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: cryptography/src/appleMain/kotlin/com/wire/kalium/cryptography/MLSClientImpl.kt [hunk 1]
```diff
@@ -84,9 +84,10 @@ class MLSClientImpl(
         return toCommitBundle(coreCrypto.joinByExternalCommit(toUByteList(publicGroupState), defaultGroupConfiguration))
     }
 
-    override suspend fun mergePendingGroupFromExternalCommit(groupId: MLSGroupId) {
+    override suspend fun mergePendingGroupFromExternalCommit(groupId: MLSGroupId): List<DecryptedMessageBundle>? {
         val groupIdAsBytes = toUByteList(groupId.decodeBase64Bytes())
         coreCrypto.mergePendingGroupFromExternalCommit(groupIdAsBytes)
+        return null
```
Comment (mchenani) [inline cryptography/src/appleMain/kotlin/com/wire/kalium/cryptography/MLSClientImpl.kt:90]: why null?

Comment (typfel) [reply cryptography/src/appleMain/kotlin/com/wire/kalium/cryptography/MLSClientImpl.kt:90]: This the apple implementation which is stuck at an old CoreCrypto version which doesn't have this new API.

**Review by mchenani — APPROVED** [review]
🛸

**Review by vitorhugods — APPROVED** [review]

**Comment by vitorhugods** [comment]
The failing Android test was while testing Crypto stuff. I'm curious to see if it was just bad luck or what. I'm triggering a new run.

**Comment by vitorhugods** [comment]
Yep. It looks like one test is stuck in a loop.

**Comment by typfel** [comment]
> Yep. It looks like one test is stuck in a loop on Android.

I think this due to a bug in the new async UniFFI bindings. There's a fix but it's not merged yet.

**Pull request merged**